src/app/patient/patient.cpp

```diff
@@ -0,0 +1,83 @@
+#include "patient.hpp"
+
+std::vector<std::shared_ptr<Patient>> Patient::patients_;
+
+Patient::Patient(const std::string name, const std::string last_name, const std::string pesel,
+                 const std::unordered_set<Allergent> allergents)
+    : Person(name, last_name, pesel), allergents_{allergents}, debt_{}
+{
+}
+
+void Patient::createPatient(const std::string name, const std::string last_name, const std::string pesel,
+                            const std::unordered_set<Allergent> allergents)
+{
+    auto patient = std::shared_ptr<Patient>(new Patient(name, last_name, pesel, allergents));
+    patients_.push_back(patient);
+}
+
+std::shared_ptr<Patient> Patient::getPatient(const std::string pesel)
+{
+    const auto patient =
+        std::find_if(patients_.begin(), patients_.end(),
+                     [&pesel](const std::shared_ptr<Patient> patient) { return patient->getPesel() == pesel; });
+
+    if (patient != patients_.end())
+    {
+        return *patient;
+    }
+
+    return nullptr;
+}
+
+std::shared_ptr<Patient> Patient::getPatient(const std::string name, const std::string last_name)
+{
+    const auto patient =
+        std::find_if(patients_.begin(), patients_.end(), [&name, &last_name](const std::shared_ptr<Patient> patient) {
+            return patient->getFirstName() == name && patient->getLastName() == last_name;
+        });
+
+    if (patient != patients_.end())
+    {
+        return *patient;
+    }
+
+    return nullptr;
+}
```

Comment:
Please fix this function, so when we have more than one patient with the same first and last name, it'll return some sort of container with pointers to these objects.

Reply:
applied